### PR TITLE
Improve dockerfile and add container tests

### PR DESCRIPTION
**Pull request by x-Xymos**
This copies the necessary libraries from the base image for rustla2 to run which reduces the image size from 4gb to around 250mb

### CODE REVIEW

File: api/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:16.04 AS build-env
+FROM ubuntu:16.04 AS base
```
Comment (jbpratt78) [inline api/Dockerfile:1]: Can we bump this version, or will it break the Dockerfile?

Comment (x-Xymos) [reply api/Dockerfile:1]: Probably, I would test it locally just to be sure

**Comment by jbpratt78** [comment]
Currently failing to build, per conversation we need to add `DOCKER_BUILDKIT=1` to the GHA file.

File: api/Dockerfile (hunk 1 above)
Comment (x-Xymos) [reply api/Dockerfile:1]: I tried to bump it up to 18.04 but the build fails

`17 40.05 [100%] Linking CXX executable rustla2_api
#17 42.20 /usr/local/lib/libfolly.a(NestedCommandLineApp.cpp.o): In function `folly::NestedCommandLineApp::NestedCommandLineApp(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, boost::program_options::variables_map const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)>)':
#17 42.20 /api/third-party/folly/folly/experimental/NestedCommandLineApp.cpp:64: undefined reference to `boost::program_options::options_description::m_default_line_length'
#17 42.20 /api/third-party/folly/folly/experimental/NestedCommandLineApp.cpp:64: undefined reference to `boost::program_options::options_description::m_default_line_length'
#17 42.20 /api/third-party/folly/folly/experimental/NestedCommandLineApp.cpp:64: undefined reference to `boost::program_options::options_description::options_description(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int, unsigned int)'
#17 42.20 /api/third-party/folly/folly/experimental/NestedCommandLineApp.cpp:86: undefined reference to `boost::program_options::options_description::add_options()'
#17 42.20 /api/third-party/folly/folly/experimental/NestedCommandLineApp.cpp:88: undefined reference to `boost::program_options::options_description_easy_init::operator()(char const*, char const*)'
#17 42.20 /api/third-party/folly/folly/experimental/NestedCommandLineApp.cpp:88: undefined reference to `boost::program_options::options_description_easy_init::operator()(char const*, char const*)'
#17 42.20 /usr/local/lib/libfolly.a(NestedCommandLineApp.cpp.o): In function `folly::NestedCommandLineApp::addCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::function<void (boost::program_options::variables_map const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)>)':
`

Comment (jbpratt78) [reply api/Dockerfile:1]: @slugalisk

Comment (x-Xymos) [reply api/Dockerfile:1]: Here's the full trace https://pastebin.com/raw/BU0j3Lgs probably something to do with the version of the compiler being different on ubuntu 18.04

Comment (x-Xymos) [reply api/Dockerfile:1]: -- The C compiler identification is GNU 7.5.0                                                                                                                                               
-- The CXX compiler identification is GNU 7.5.0 
-- Boost version: 1.65.1

-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Boost version: 1.58.0

**Comment by x-Xymos** [comment]
> Currently failing to build, per conversation we need to add `DOCKER_BUILDKIT=1` to the GHA file.

I actually tried it again both with docker buildkit enabled and without and they both build for me so I'm not sure if this fixes it or not.

File: api/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 
 RUN apt-get update && apt-get install -y \
     automake \
     autoconf \
-    autoconf-archive \
+     autoconf-archive \
```
Comment (slugalisk) [inline api/Dockerfile]: typo

**Pull request merged**